src/cpp/rtps/history/History.cpp

```diff
@@ -47,6 +47,41 @@ History::~History()
     logInfo(RTPS_HISTORY, "");
 }
 
+History::const_iterator History::find_change(
+        CacheChange_t* ch)
+{
+    if (mp_mutex == nullptr)
+    {
+        logError(RTPS_HISTORY, "You need to create a RTPS Entity with this History before using it");
+        return const_iterator();
```

Comment:
The `const_iterator()` should be set to `changesEnd()`.

Reply:
this way I assure that lack of mutex leads to assertion on next iterator check (at least for Visual Studio and Debug config) 😎😎